apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/VedtaksbrevService.kt

```diff
@@ -90,29 +91,42 @@ class VedtaksbrevService(
         val behandling = sakOgBehandlingService.hentBehandling(brev.sakId, brev.behandlingId!!, brukerTokenInfo)
         val avsender = adresseService.hentAvsender(behandling.vedtak)
 
-        val (brevKode, brevData) = opprettBrevData(brev, behandling)
-        val brevRequest = BrevbakerRequest.fra(brevKode, brevData, behandling, avsender)
+        val kode = BrevDataMapper.brevKode(behandling, brev.prosessType)
+        val brevData = opprettBrevData(brev, behandling)
+        val brevRequest = BrevbakerRequest.fra(kode.ferdigstilling, brevData, behandling, avsender)
 
         return genererPdf(brev.id, brevRequest)
             .also { pdf -> ferdigstillHvisVedtakFattet(brev, behandling, pdf, brukerTokenInfo) }
     }
 
-    private fun opprettBrevData(brev: Brev, behandling: Behandling): Pair<EtterlatteBrevKode, BrevData> =
+    private fun opprettBrevData(brev: Brev, behandling: Behandling): BrevData =
         when (brev.prosessType) {
-            AUTOMATISK -> BrevDataMapper.fra(behandling)
-            MANUELL -> {
-                val payload = requireNotNull(db.hentBrevPayload(brev.id))
-
-                // TODO: Map brevkode
-                Pair(EtterlatteBrevKode.OMS_OPPHOER_MANUELL, ManueltBrevData(payload.elements))
+            AUTOMATISK -> {
+                when (behandling.revurderingsaarsak) {
+                    RevurderingAarsak.OMGJOERING_AV_FARSKAP -> manueltBrevData(brev)
+                    RevurderingAarsak.ADOPSJON -> manueltBrevData(brev)
+                    else -> BrevDataMapper.brevData(behandling)
+                }
             }
+            MANUELL -> manueltBrevData(brev)
         }
 
-    private fun opprettInnhold(behandling: Behandling, prosessType: BrevProsessType): BrevInnhold {
+    private fun manueltBrevData(brev: Brev) = ManueltBrevData(requireNotNull(db.hentBrevPayload(brev.id)).elements)
+
+    private suspend fun opprettInnhold(behandling: Behandling, prosessType: BrevProsessType): BrevInnhold {
         val tittel = "Vedtak om ${behandling.vedtak.type.name.lowercase()}"
 
         val payload = when (prosessType) {
```

Comment:
Hadde det gitt mening med en tredje prosesstype for brev? I denne og forrige `when` er det basically en sjekk på om du er en spesiell versjon av automatisk i kraft av revurderingsårsak, men det kunne vært basert på f.eks. prosessType `REDIGERBAR_AUTOMATISK` e.l.

Har ikke oversikt over de andre stedene det evt. må håndteres da :p

Reply:
Oh, det er ein fristande ide. Samtidig trur eg at vi etter kvart skal over på redigerbar automatisk for alle brev, så usikker på om det vil trengs på sikt 🤔